client/src/search/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,577 @@
+
+      return is_on_displayed_page;
+    };
+
+    const paginated_results = _.filter(
+      queried_results,
+      (queried_result, index) => paginate_results(queried_result, index)
+    );
+
+    const page_range_start = pagination_index * pagination_size + 1;
+    const page_range_end = page_range_start + paginated_results.length - 1;
+
+    const total_matching_results = queried_results.length;
+
+    const remaining_results =
+      total_matching_results - (pagination_index + 1) * pagination_size;
+    const next_page_size =
+      remaining_results < pagination_size ? remaining_results : pagination_size;
+
+    const needs_pagination_up_control = pagination_index > 0;
+    const needs_pagination_down_control =
+      page_range_end < total_matching_results;
+
+    const status_props = {
+      current_selected:
+        paginated_results[current_selected_index] &&
+        paginated_results[current_selected_index].name,
+      current_selected_index,
+      min_length,
+      total_matching_results,
+      page_range_start,
+      page_range_end,
+      query_length: search_text.length,
+      input_is_in_focus,
+      needs_pagination_up_control,
+      needs_pagination_down_control,
+      pagination_size,
+      next_page_size,
+      paginated_results,
+    };
+
+    return (
+      <div className="rbt" style={{ position: "relative" }} ref={this.rbtRef}>
+        <div className="search-bar">
+          <div className="search-icon-container">
+            <span aria-hidden="true">
+              <img
+                src={`${get_static_url("svg/search.svg")}`}
+                style={{ width: "30px", height: "30px" }}
+              />
+            </span>
+          </div>
+          <input
+            id="rbt-search"
+            role="combobox"
+            autoComplete="off"
+            aria-autocomplete="list"
+            aria-owns={this.menuId}
+            aria-expanded={show_menu}
+            style={{ width: "100%" }}
+            placeholder={placeholder}
+            onChange={update_search_text}
+            ref={this.typeaheadRef}
+            value={search_text}
+            onFocus={() =>
+              this.setState({ can_show_menu: true, input_is_in_focus: true })
+            }
+            onBlur={() => this.setState({ input_is_in_focus: false })}
+            onKeyDown={(e) =>
+              this.handle_key_down(
+                e,
+                show_menu,
+                paginated_results,
+                needs_pagination_up_control,
+                needs_pagination_down_control
+              )
+            }
+          />
+
+          {filter_content && (
+            <DropdownMenu
+              dropdown_trigger_txt={
+                <div
+                  style={{
+                    textAlign: "start",
+                    whiteSpace: "nowrap",
+                    display: "inline-block",
+                  }}
+                >
```

Comment:
Okay, contemplated if I should make `utility_buttons` a prop inside `AdvancedSearch` but decided it wouldn't be useful since it needs state management at AdvancedSearch level